src/GitTfs/Core/IGitRepository.cs

```diff
@@ -62,5 +62,7 @@ public interface IGitRepository : IGitHelpers
         IEnumerable<GitCommit> FindParentCommits(string fromCommit, string toCommit);
         bool IsPathIgnored(string relativePath);
         string CommitGitIgnore(string pathToGitIgnoreFile);
+
+        IDictionary<string, string> GetCommitChangeSetPairs();
```

Comment:
Confused about the typing here. The Value type (string) seems valid, as that's probably the best type to store a git commit-hash in, but I feel the Key type (also string) storing changeset-ids could be better captured with something like int...

Reply:
yes, using an int makes more sense.